src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseBuildConfigRepository.java

```diff
@@ -0,0 +1,34 @@
+package de.tum.in.www1.artemis.repository;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_CORE;
+
+import java.util.Optional;
+
+import org.hibernate.Hibernate;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Repository;
+
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.domain.ProgrammingExerciseBuildConfig;
+import de.tum.in.www1.artemis.repository.base.ArtemisJpaRepository;
+
+@Profile(PROFILE_CORE)
+@Repository
+public interface ProgrammingExerciseBuildConfigRepository extends ArtemisJpaRepository<ProgrammingExerciseBuildConfig, Long> {
+
+    Optional<ProgrammingExerciseBuildConfig> findByProgrammingExerciseId(Long programmingExerciseId);
+
+    default ProgrammingExerciseBuildConfig getProgrammingExerciseBuildConfigElseThrow(ProgrammingExercise programmingExercise) {
+        if (programmingExercise.getBuildConfig() == null || !Hibernate.isInitialized(programmingExercise.getBuildConfig())) {
+            return getValueElseThrow(findByProgrammingExerciseId(programmingExercise.getId()));
+        }
+        return programmingExercise.getBuildConfig();
+    }
+
+    default void generateBuildPlanAccessSecretIfNotExists(ProgrammingExerciseBuildConfig buildConfig) {
+        if (!buildConfig.hasBuildPlanAccessSecretSet()) {
+            buildConfig.generateAndSetBuildPlanAccessSecret();
+            save(buildConfig);
+        }
+    }
```

Comment:
Why is this a repository-method? Seems better fitted for a Service.

Reply:
I couldnt move it to `ProgrammingExerciseService` since there would be a circular dependency with gitlab/jenkins services. And creating a new service just for 1 method seems overkill. So i would keep it there for now.